[SPARK-10718] [BUILD] Update License on conf files and corresponding excludes file update

rekhajoshm opened the pull request:
Update License on conf files and corresponding excludes file update

rekhajoshm committed:
Merge pull request #1 from apache/master

Pulling functionality from apache spark

rekhajoshm committed:
Merge pull request #2 from apache/master

pull latest from apache spark

rekhajoshm committed:
Merge pull request #3 from apache/master

Pulling functionality from apache spark

rekhajoshm committed:
Merge pull request #4 from apache/master

Pulling functionality from apache spark

rekhajoshm committed:
updated to exclude conf files

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #42725 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42725/consoleFull) for   PR 8842 at commit [`6285eb3`](https://github.com/apache/spark/commit/6285eb39ba508a6a91ad42e66ab74ed93298a267).

SparkQA commented:
[Test build #42725 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42725/console) for   PR 8842 at commit [`6285eb3`](https://github.com/apache/spark/commit/6285eb39ba508a6a91ad42e66ab74ed93298a267).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42725/
Test FAILed.

rekhajoshm commented:
retest please

rxin commented on .rat-excludes:
```diff
@@ -15,15 +15,15 @@ TAGS
+metrics.properties.*
 slaves
-slaves.template
+slaves.*
```
this is casting a net too wide -- i.e. one day we might have slaves.scala and this won't catch it.

srowen replied:
Yeah, per the JIRA, if the intent is to exclude `conf/spark-defaults.conf`, that's not actually in the distribution.

rxin commented:
@rekhajoshm why do you need this change? This file is mostly run by Jenkins.

rekhajoshm commented:
thanks @rxin  and @srowen for your quick review comments. @rxin i need it for ./dev/run-tests to run correctly. imo nothing under conf/ folder needs to be checked for apache license.most of the users will convert .template files to actual conf files., and will need to replicate this at their end to make sure run-tests work.

srowen commented:
Running tests including things like RAT checks is usually for preparing a packaging of Spark. If you have `spark-defaults.conf` in there then you are also running a local Spark cluster or something, which is a different concern? I don't think these should happen from the same dir. 

Conf files and templates are part of the copyrightable content of the project, so do technically need the same licensing regime, so I don't agree that `conf/` should be completely skipped. However `spark-defaults.conf` is not a product of the project. That seems to be the only file at issue. If it exists it was user created, and could be skipped. But then why is the change here skipping other stuff? it's not even skipping checks of `spark-defaults.conf`

rekhajoshm commented:
Ok @srowen .If you both review, validate its not correct/needed, thats fine. @rxin

imo more than how the configuration is done or whether call should ever be invoked by user/developer, its a matter of consistency.None of the conf/ files have the apache license atm. Maybe they all can include it then?If not then all files under conf/ can forgo check on license.
@srowen You are right. I have a local setup with conf files which I use for tinkering, need run-tests to be working for pull request.You can verify this pull makes the change spark-defaults.conf.* which works.There are additional similar changes.

srowen commented:
Yeah I'm mixing up a few things here. You're right none of them do have header files. I think the template files should have the same header IMHO unless there's some reason they can't contain one, or maybe someone had a good reason that they don't need it. Maybe that's a separate point.

Skipping `spark-defaults.conf` isn't a problem I think, since it would not be from the project anyway. If it's a developer convenience, OK, even if you probably wouldn't normally have this in your distro directory if you're packaging Spark.

But yeah only that file should be skipped too, to keep it conservative. `spark-defaults.conf.*` didn't look loike it matched `spark-defaults.conf`, but, aha, this file is actually intended to contain regexes? then it does yes. Then, lots of entries in this file aren't quite expressing what they  mean to, like "service.properties" or "test_support/*" or arguably ".*iws". Then I see a bunch of entries that probably shouldn't be there like "local-1422981759269/*"

So, I propose that this change at least only affect `spark-defaults.conf` but if you're up for it, I think there is significant additional cleanup that could be done here!

rekhajoshm committed:
updated for review comments

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

rekhajoshm committed:
updated as now conf files have apache licenses

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #42735 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42735/consoleFull) for   PR 8842 at commit [`959e1cb`](https://github.com/apache/spark/commit/959e1cbddabf388bfec04d1cf10962d559634e45).

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42734/
Test FAILed.

rekhajoshm commented:
Thanks @srowen for your comments.

While it is regex, bit different, the other folders you highlight needs exclusion.For example local-* are test resource files under spark/core/src/test/resources/spark-events/.As per your review comments, I have added apache license for all files under conf/ folder and few others, and updated excludes file accordingly.Also as suggested by @rxin and I agree slaves is not a candidate for flexible exclusion.updated.The build had passed locally.Please review.
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 09:28 min
[INFO] Finished at: 2015-09-20T14:47:50-07:00
[INFO] Final Memory: 395M/1723M
```

SparkQA commented:
[Test build #42735 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42735/console) for   PR 8842 at commit [`959e1cb`](https://github.com/apache/spark/commit/959e1cbddabf388bfec04d1cf10962d559634e45).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/42735/
Test PASSed.

srowen commented:
I support this change, which amounts to adding licenses to files that didn't have it before. Now, if someone cloned `spark-defaults.conf` it would actually pass RAT checks anyway. I wouldn't mind adding it to the excludes if you like (yes, coming around to your original point now) but it should Just Work anyway now.

Correcting the syntax of the excludes file could wait for another issue I suppose. Let me let this stay open another day or so for comments but I think it's a good change and solves your original issue.

rekhajoshm commented:
sure, updated pull title too.thanks @srowen !

rxin commented:
@rekhajoshm can you update the pull request description itself?

rekhajoshm commented:
Done @rxin thanks.

srowen commented:
Merged to master
